.github/workflows/sync-with-PRO.jsonnet

```diff
@@ -13,10 +15,63 @@ local gha = import 'libs/gha.libsonnet';
 
 local job = {
   'runs-on': 'ubuntu-latest',
-  steps: [
+  permissions: gha.write_permissions,
+  steps: semgrep.github_bot.get_token_steps + [
      {
-       run: 'echo TODO'
+      name: 'Checkout OSS',
+      uses: 'actions/checkout@v3',
+      with: {
+        ref: '${{ github.event.repository.default_branch }}',
+        // fetch all history
+        'fetch-depth': 0,
+        // Use the token provided by the JWT token getter above
+        token: semgrep.github_bot.token_ref,
+      },
      },
+     {
+      name: 'Checkout PRO',
+      uses: 'actions/checkout@v3',
+      with: {
+        repository: 'semgrep/semgrep-proprietary',
+        path: 'PRO',
+        token: semgrep.github_bot.token_ref,
+       },
+     },
+     {
+      name: 'Creating the branch and commiting to it',
+      env: {
+        BRANCHNAME: 'sync-with-PRO-x3-${{ github.run_id }}-${{ github.run_attempt }}',
+        GITHUB_TOKEN: semgrep.github_bot.token_ref,
+      },
+       run: |||
+         # will generate a 0001-xxx patch
+         git format-patch develop^
+         cd PRO
+         git config --global user.name "GitHub Actions Bot"
+         git config --global user.email "<>"
+         git checkout -b $BRANCHNAME
+         #TODO: apply patch from OSS HEAD to this branch
+         cd OSS
+         # note that this can fail if develop^ is already in Pro
+         # in fact patch will propose to use -R to revert the patch instead
+         patch -p1 < ../../0001-*
+         # TODO: use git am to keep the subject/author/etc of the 0001-xxx
```

Comment:
Why making this a TODO? It's pretty easy and it's not nice to loose author info in Pro.

```bash
sed -E 's:( a| b)/:\1OSS/:g' 0001-*.patch >pro.patch
git am pro.patch
```

Reply:
ok will add, thx

Reply:
when I have something working in GHA, even if it's not perfect, I want to commit it ASAP because you never know what little change can affect the thing and suddenly make the workflow fail ...

Reply:
Nice :)

Reply:
You're getting addicted!